src/main/java/io/jenkins/plugins/gitlabbranchsource/helpers/GitLabPipelineStatusNotifier.java

```diff
@@ -207,10 +207,21 @@ private static void logComment(Run<?, ?> build, TaskListener listener) {
     /**
      * Retrieves the source project ID for a merge request
      */
-    private static Integer getSourceProjectId(Job job, GitLabApi gitLabApi, String projectPath) {
+    static Object getSourceProjectId(Job job, GitLabApi gitLabApi, String projectPath) {
         LOGGER.log(Level.INFO, "Getting source project ID from MR");
         String[] jobFullNameParts = job.getFullName().split("-");
-        Integer mrId = Integer.parseInt(jobFullNameParts[jobFullNameParts.length - 1]);
+        Integer mrId = null;
+        for (int i = jobFullNameParts.length - 1; i >= 0; i--) {
+            try {
+                mrId = Integer.parseInt(jobFullNameParts[i]);
+                break;
+            } catch (NumberFormatException e) {
+            }
+        }
```

Comment:
Why is this necessary? Split with limit should do the trick? unless there is something odd going odd here.

```java
        String[] jobFullNameParts = job.getFullName().split("-", 2);
```

Reply:
I did not change that line...

Reply:
Hmm not sure what to make of it than 😵

Reply:
also check issue #164 for the details, my jobs are named `MR-123-head` and `MR-123-merge`, so yes I could assume if it is not the last part it will be the previous one, but I though it'd be safer to not assume (again) that project names will always follow that naming convention. As I understand that comes from another plugin so by being more tolerant it might work if something changes.

Reply:
well version `1.5.9` introduced that logic where it gets the project ID from the MR id. Logic that did not consider people that choose the build strategy `Both the current merge request revision and the merge request merged with the current target branch revision`.

Now I'm trying to fix that by:
1. Parse the project name differently, not to assume the last part of the project name is the MR number
2. If for whatever reason we cannot get the project id, revert to the old logic (of `1.5.8`) of using the project name instead.

Reply:
These names come from this plugin, so we know the format.

```java
String[] jobFullNameParts = job.getFullName().split("-", 3);
Integer mrId = jobFullNameParts.length == 3 ? jobFullNameParts[1] : jobFullNameParts[1];
```

Reply:
This is the commit that broke it for us: https://github.com/jenkinsci/gitlab-branch-source-plugin/commit/cbe0d50c4b628bc669864403a651159ec56db8a7